### PR TITLE
Add __str__ method for Dataset

### PR DESCRIPTION
**Pull request by JouvinLea**
This PR start implementation of __str__ method for dataset class related to this issue (https://github.com/gammapy/gammapy/issues/2234)

I started with the fluxpoint dataset. Not sure about where to put the test. Once this is ok, we continue for the others dataset

### CODE REVIEW

File: gammapy/spectrum/flux_point.py [hunk 1]
```diff
@@ -1187,6 +1187,29 @@ def __init__(self, model, data, mask_fit=None, likelihood="chi2", mask_safe=None
...
+        str_ += "Total Number of flux points : {} ".format(len(self.data.table))
+        str_ += "where {} were excluded for safe energy range ".format(
+            len(np.where(self.mask_safe == False)[0])
+        )
+        if self.mask_fit is not None:
+            str_ += "and {} were excluded by the user".format(
+                len(np.where(self.mask_fit == False)[0])
+            )
+        str_ += "\n"
+        str_ += "likelihood value   : {}\n".format(len(self._likelihood))
```
Comment (registerrier) [inline gammapy/spectrum/flux_point.py]: Can this fail if no model has been set?
Should there be an if statement to check for the presence of a model before trying to print the likelihood value?

**Comment by JouvinLea** [comment]
@adonath @registerrier 
Thanks for your comments! You can try it out for FluxDataset and MapDataset

**Comment by JouvinLea** [comment]
Don't try SpetrumDataset I will improve it now!

**Comment by JouvinLea** [comment]
@adonath @registerrier 
Everything implemented, you can do the final review

**Comment by adonath** [comment]
Thanks @JouvinLea, I have made a few clean-up commits. Will merge now...

**Pull request merged**